### ISSUE TITLE
Cluster Tools: 'Install' button displays after installing chart when running Rancher locally

### ISSUE THREAD

**Issue by yonasberhe23** [rancher/dashboard#9940]
**Setup**
- Rancher version: v2.8.0
- Rancher UI Extensions:
- Browser type & version: Any

**Describe the bug**
The 'Install' button is displayed after installing a chart when running Rancher locally. Edit and delete buttons should be displayed instead. It seems as though Rancher does not recognize that the installation occurred when running locally.

**To Reproduce**
- run Rancher instance locally
- login to Rancher as an admin
- go to Cluster Tools page (https://localhost:8005/c/local/explorer/tools)
- install chart ('Alerting Drivers')
- after install notice edit and delete buttons are missing from the chart card 
- go to remote setup (http://yonasb.qa.rancher.space/c/local/explorer/tools)
- notice edit and delete buttons are displayed on the chart card 

**Result**
'Install' button is displayed after installing a chart when using localhost.

**Expected Result**
Edit and delete buttons display after installing a chart when using localhost.

**Screenshots**
...show what is displayed in localhost vs. remote setup after chart installation
<img width="1460" alt="Screenshot 2023-10-17 at 10 22 56 AM" src="https://github.com/rancher/dashboard/assets/127343932/187fc1fd-d0da-4739-8d01-6ae12d049c33">
<img width="1465" alt="Screenshot 2023-10-17 at 10 09 54 AM" src="https://github.com/rancher/dashboard/assets/127343932/f52b249e-b131-46ab-a42e-566c61139af3">

**Additional context**

**Comment by richard-cox** [comment]
This issue blocks incomming e2e tests https://github.com/rancher/dashboard/pull/9749

**Comment by torchiaf** [comment]
The is rarely replicable, but I noticed that, just before to install a new App from catalog, we receive some errors from `catalog.cattle.io.app` websocket; at least in my env, those kind of ws errors are very frequent.


![Screenshot from 2024-02-29 17-10-12](https://github.com/rancher/dashboard/assets/26394656/36c02b75-2509-46ea-86a4-e7f14b1958e3)

In this case, the UI loses the new messages containing the new app list.

I pushed a [draft](https://github.com/rancher/dashboard/pull/10536) to re-sync those resource having failed ws messages, ONLY to try the e2e tests and analyze the issue, but I would be very careful in touching the ws utils.

@richard-cox do you have any idea on this?

**Comment by richard-cox** [comment]
Those are frustrating warnings sent when the BE closes a steve socket (there's an issue somewhere for them to fix). So that would track what you're seeing.

When a socket goes down we should immediately re-subscribe. Any resource that was previously being watched should re-watch with a `revision` of the last updates resource. Any changes that happen after that should in theory be re-sent over that socket. 

Does the socket always re-connect ok, and does the new connection have any messages containing the resource in question?

**Comment by torchiaf** [comment]
> Those are frustrating warnings sent when the BE closes a steve socket (there's an issue somewhere for them to fix). So that would track what you're seeing.
> 
> When a socket goes down we should immediately re-subscribe. Any resource that was previously being watched should re-watch with a `revision` of the last updated resource (ish). Any changes that happen after was changed should in theory be re-sent over that socket.
> 
> Does the socket always re-connect ok, and does the new connection have any messages containing the resource in question?

For what i've seen the UI doesn't receive the lost updates for `catalog.cattle.io.app`.
Adding this check https://github.com/rancher/dashboard/blob/e80026181283ab07d11041d7ad856bb925f80d81/shell/plugins/steve/subscribe.js#L850
fixed the issue and the e2e tests, since we do a `re-sync` for this type of ws errors, but I'm not sure if this is the right way to fix it until we don't have a fix back-end side.

**Comment by richard-cox** [comment]
That works around the issue and would cause a LOT of http requests to fetch all resources.

If the BE isn't sending the correct updates after re-watching on the new socket connection with an old revision then you'll need to create an issue `rancher/rancher` side to address. It'll need some clear detail on the socket messages, specifically the ones we're getting and the one's we're missing.

**Comment by torchiaf** [comment]
> That works around the issue and would cause a LOT of http requests to fetch all resources.
> 
Exactly; I'll collect more info to send a t the backend team. Thanks
